crates/query-engine/translation/src/translation/query/mod.rs

```diff
@@ -30,19 +30,15 @@ pub fn translate(
         None,
         variables_table_ref,
     );
-    let (current_table, from_clause) = root::make_from_clause_and_reference(
-        &query_request.collection,
-        &query_request.arguments,
-        &env,
-        &mut state,
-        None,
-    )?;
 
-    let select_set = translate_query(
+    let select_set = root::translate_query(
         &env,
         &mut state,
-        &current_table,
-        &from_clause,
+        root::MakeFrom::Collection {
```

Comment:
Instead of building a from clause and passing it, we build a structure describing how to build the from clause, so it can be built the same way for both `rows` and `aggregates` but produce different aliases using the state index.